src/lib.rs

```diff
@@ -51,15 +51,46 @@ impl Pool {
         pool
     }
 
+    /// Create a new Pool with `size` threads and given thread config.
+    ///
+    /// If `size` is zero, no threads will be spawned. Threads can
+    /// be added later via `expand`.
+    ///
+    /// NOTE: Since Pool can be freely cloned, it does not represent a unique
+    /// handle to the thread pool. As a consequence, the thread pool is not
+    /// automatically shut down; you must explicitly call `Pool::shutdown` to
+    /// shut down the pool.
+    #[inline]
+    pub fn with_thread_config(size: usize, thread_config: ThreadConfig) -> Pool {
+        // Create an empty pool with configuration.
+        let pool = Pool {
```

Comment:
Sorry for not catching this in the first round of review, but there's some slight duplication here between this function and `Pool::empty` that could be addressed by deriving default for `ThreadConfig` and deriving or implementing (if you want to keep the starting thread number at 1 instead of 0) `Default` for `PoolInner`, then using that implementation both here and in `Pool::empty`. I'm concerned that it would be too easy to update either `empty` or `with_thread_config` and not the other.